packages/react-dom/src/shared/DOMProperty.js

```diff
@@ -226,6 +221,23 @@ const properties = {};
   );
 });
 
+// For legacy purposes, defaultValue stringifies boolean values
```

Comment:
How old is this behavior?
Would be nice to make this comment more specific than "legacy purposes".
Do we want to change it?

Reply:
As long as I can remember. All reserved props get passed through without any validation from `UnknownDOMPropertyHook`. 

How often do people pass `true` and `false`? It seems like it would rarely come up for `input`s, but often enough for `select`s. What do you think?

Also I can work on making the comment better :)